WordPress-iOS-Shared/Core/WPImageSource.m

```diff
@@ -24,17 +25,19 @@ + (instancetype)sharedSource
 
 - (void)dealloc
 {
-    [_downloadingQueue cancelAllOperations];
+    [_downloadsSession invalidateAndCancel];
 }
 
 - (id)init
 {
     self = [super init];
     if (self) {
-        _downloadingQueue = [[NSOperationQueue alloc] init];
         _urlDownloadsInProgress = [[NSMutableSet alloc] init];
         _successBlocks = [[NSMutableDictionary alloc] init];
         _failureBlocks = [[NSMutableDictionary alloc] init];
+
+        NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+        _downloadsSession = [NSURLSession sessionWithConfiguration:configuration];
```

Comment:
Just wondering if we don't wanna use the backgroundSessionConfiguration, which would allow us to finish downloading the assets, even when the app dies. Or not! just a thought